internal/state/relationship/capturer.go

```diff
@@ -63,27 +83,78 @@ func (c *CapturerImpl) Capture(obj client.Object) {
 				Name:      svcName,
 			}
 		}
+	case *v1beta1.Gateway:
+		var selectors []labels.Selector
+		for _, listener := range o.Spec.Listeners {
+			if selector := graph.GetAllowedRouteLabelSelector(listener); selector != nil {
+				convertedSelector, err := metav1.LabelSelectorAsSelector(selector)
+				if err == nil {
+					selectors = append(selectors, convertedSelector)
+				}
+			}
+		}
+
+		gatewayName := client.ObjectKeyFromObject(o)
+		if len(selectors) > 0 {
+			c.gatewayLabelSelectors[gatewayName] = selectors
+			for ns, cfg := range c.referencedNamespaces {
+				var gatewayMatches bool
+				for _, selector := range selectors {
+					if selector.Matches(labels.Set(cfg.labelMap)) {
+						gatewayMatches = true
+						cfg.match = true
+						cfg.gateways[gatewayName] = struct{}{}
+						break
+					}
+				}
+				if !gatewayMatches {
+					// gateway labels have changed, so remove the namespace relationship
+					c.removeGatewayReferenceFromNamespaces(gatewayName, referencedNamespaces{ns: cfg})
+				} else {
+					c.referencedNamespaces[ns] = cfg
+				}
+			}
+		} else if _, exists := c.gatewayLabelSelectors[gatewayName]; exists {
+			// label selectors existed previously for this gateway, so clean up any references to them
+			c.removeGatewayLabelSelector(gatewayName)
+		}
+	case *v1.Namespace:
+		nsLabels := o.GetLabels()
+		gateways := c.matchingGateways(nsLabels)
+		nsCfg := namespaceCfg{
+			labelMap: nsLabels,
+			match:    len(gateways) > 0,
```

Comment:
is match field necessary? can we just use a condition `len(gateways) > 0` instead or have a method that uses that condition?  Or have My concern: having duplicated data (multiple sources of truths).